sqlobject/src/main/java/org/jdbi/v3/sqlobject/BatchHandler.java

```diff
@@ -101,9 +109,84 @@ private int indexOfBatchChunkSizeParameter(Method method)
     @Override
     public Object invoke(Object target, Method method, Object[] args, HandleSupplier h)
     {
-        boolean foundIterator = false;
-        Handle handle = h.getHandle();
+        final Handle handle = h.getHandle();
+        final String sql = handle.getConfig().get(SqlObjects.class)
+                .getSqlLocator().locate(sqlObjectType, method);
+        final int chunkSize = batchChunkSize.call(args);
+        final Iterator<Object[]> batchArgs = zipArgs(args);
+
+        ResultIterator<Object> result = new ResultIterator<Object>() {
+            ResultIterator<?> batchResult;
+
+            {
+                hasNext(); // Ensure our batchResult is prepared, so we can get its context
+            }
+
+            @Override
+            public boolean hasNext() {
+                // first, any elements already buffered?
+                if (batchResult != null) {
+                    if (batchResult.hasNext()) {
+                        return true;
+                    }
+                    // no more in this chunk, release resources
+                    batchResult.close();
+                }
+                // more chunks?
+                if (!batchArgs.hasNext()) {
+                    return false;
+                }
+                // execute a single chunk and buffer
+                PreparedBatch batch = handle.prepareBatch(sql);
+                applyCustomizers(batch, args);
+                for (int i = 0; i < chunkSize && batchArgs.hasNext(); i++) {
+                    applyBinders(batch.add(), batchArgs.next());
+                }
+                batchResult = executeBatch(handle, batch);
+                return hasNext(); // recurse to ensure we actually got elements
+            }
+
+            @Override
+            public Object next() {
+                if (!hasNext()) {
+                    throw new NoSuchElementException();
+                }
+                return batchResult.next();
+            }
+
+            @Override
+            public StatementContext getContext() {
+                return batchResult.getContext();
+            }
+
+            @Override
+            public void close() {
+                batchResult.close();
```

Comment:
Nit: We should probably remember when this method is called so that successive calls to the iterator don't continue executing more batches.